addons/sale_stock/models/sale_order.py

```diff
@@ -224,13 +224,24 @@ def _prepare_procurement_values(self, group_id=False):
     def _get_qty_procurement(self):
         self.ensure_one()
         qty = 0.0
-        for move in self.move_ids.filtered(lambda r: r.state != 'cancel'):
-            if move.picking_code == 'outgoing':
-                qty += move.product_uom._compute_quantity(move.product_uom_qty, self.product_uom, rounding_method='HALF-UP')
-            elif move.picking_code == 'incoming':
-                qty -= move.product_uom._compute_quantity(move.product_uom_qty, self.product_uom, rounding_method='HALF-UP')
+        outgoing_moves, incoming_moves = self._get_outgoing_incoming_moves()
+        for move in outgoing_moves:
+            qty += move.product_uom._compute_quantity(move.product_uom_qty, self.product_uom)
+        for move in incoming_moves:
+            qty -= move.product_uom._compute_quantity(move.product_uom_qty, self.product_uom)
         return qty
 
+    def _get_outgoing_incoming_moves(self):
+        outgoing_moves = self.env['stock.move']
+        incoming_moves = self.env['stock.move']
+        for move in self.move_ids.filtered(lambda r: r.state != 'cancel' and not r.scrapped):
+            if move.location_dest_id.usage == "customer":
+                if not move.origin_returned_move_id or (move.origin_returned_move_id and move.to_refund):
```

Comment:
This line does not make sense to me.  Should not it be similar to the stock_account incoming/outgoing?